### PR TITLE
add missing abort method for sad path

### PR DESCRIPTION
**Pull request by whamtet**
Thanks @domenic and team for the great project,

Your XMLHttpRequest implementation wraps an internal client called `client`.  When we abort the request it aborts the internal client [here](https://github.com/jsdom/jsdom/blob/master/lib/jsdom/living/xhr/XMLHttpRequest-impl.js#L364).  Now this internal client is created in `xhr-utils.createClient` which has a number of paths for data url, file url etc.  At the catch statement [here](https://github.com/jsdom/jsdom/blob/master/lib/jsdom/living/xhr/xhr-utils.js#L326) we still assign client, but it is an `EventEmitter` which lacks an abort method.  If we then try to abort the wrapping XMLHttpRequest it throws a strange exception which doesn't tell the user about the true root cause.  I suggest adding an empty abort method to prevent confusion.

I want to add a unit test as well, but I'm not sure how to trigger the catch block.  If you agree with this change I can do so using e.g. Sinon or any other way that you suggest.

### CODE REVIEW

**Comment by domenic** [comment]
Thanks for the fix!

To add a test, and to ensure this is fixing a real bug, we need to expose the bug in a web page that runs inside jsdom. See https://github.com/jsdom/jsdom/blob/master/Contributing.md#web-platform-feature-tests for how to write those.

**Comment by whamtet** [comment]
According to [Contributing.md](https://github.com/jsdom/jsdom/blob/master/Contributing.md) 

All tests for web platform features (as opposed to features of jsdom itself, such as the JSDOM() constructor) should be in [web-platform-tests](https://github.com/web-platform-tests/wpt) format.

How do you decide whether a test should be an api test or web platform test?

**Comment by domenic** [comment]
XHR is a web platform feature, so it needs a web platform test.

**Comment by whamtet** [comment]
Sorry for the delay in updating @domenic , I do think this is quite an important bugfix however.  The issue is that if we make an invalid XMLHttpRequest and then abort it, the code triggers the internal abort method [here](https://github.com/jsdom/jsdom/blob/master/lib/jsdom/living/xhr/XMLHttpRequest-impl.js#L365).  On the error path this will be the `EventEmitter` defined [here](https://github.com/jsdom/jsdom/blob/master/lib/jsdom/living/xhr/xhr-utils.js#L327).  It needs its own abort method (no-op) to prevent user confusion.

In my use case the error was triggered by a third party library, but it was throwing the exception due to the missing abort method.  It looks like I need to create a new upstream web platform test for this one, please advice if you would like me to do it differently.

File: lib/jsdom/living/xhr/xhr-utils.js [hunk 1]
```diff
@@ -327,6 +327,9 @@ function createClient(xhr) {
       const eventEmitterclient = new EventEmitter();
       process.nextTick(() => eventEmitterclient.emit("error", e));
       eventEmitterclient.end = () => {};
+      eventEmitterclient.abort = () => {
```
Comment (whamtet) [inline lib/jsdom/living/xhr/xhr-utils.js:330]: this method has to exist to be consistent with the other code paths.

**Comment by domenic** [comment]
Great, the test looks perfect. I'll try to find some jsdom time over the weekend to verify that the test fails without your patch, and then do a merge and release.

**Review by domenic — APPROVED** [review]
Thank you!

**Comment by whamtet** [comment]
Thanks @domenic!!!

**Pull request merged**